### ISSUE TITLE
Literal tuple not assignable to Collection[Literal[...]]

### ISSUE THREAD

**Issue by decorator-factory** [python/mypy#17063]
**Bug Report**

Tuple literal is not assignable to a Collection argument/variable with `Literal[...]` as type argument

**To Reproduce**

[mypy-play link](https://mypy-play.net/?mypy=master&python=3.12&flags=strict-equality&gist=d28c097b521792d68d3739f5ce953339)

```python
from collections.abc import Collection
from typing import Literal

Mode = Literal["r", "w", "rw"]

def set_modes(modes: Collection[Mode]) -> None:
    pass
    
set_modes(("r", "rw"))
```

However, `set_modes(["r", "rw"])` seems to work fine

**Unexpected error**
```
main.py:9: error: Argument 1 to "set_modes" has incompatible type "tuple[str, str]"; expected "Collection[Literal['r', 'w', 'rw']]"  [arg-type]
```

**Your Environment**

- Mypy version used: 1.9.0
- Mypy command-line flags: strict, strict-inequality (doesn't seem to matter)
- Python version used: 3.12

**Comment by decorator-factory** [comment]
Seems like the error doesn't appear in mypy `1.15.0` anymore :+1:
